tests/RocksDBEngine/IndexEstimatorTest.cpp

```diff
@@ -333,12 +344,86 @@ TEST_F(IndexEstimatorTest, test_truncate_logic_2) {
 
   auto expected = currentSeq;
   std::string serialization;
-  est.serialize(serialization, ++currentSeq);
+  est.serialize(serialization, ++currentSeq, format);
   serialization.clear();
   ASSERT_EQ(est.appliedSeq(), expected);
   ASSERT_EQ(1.0, est.computeEstimate());
 
-  est.serialize(serialization, ++currentSeq);
+  est.serialize(serialization, ++currentSeq, format);
   ASSERT_EQ(est.appliedSeq(), expected);
   ASSERT_EQ(1.0, est.computeEstimate());
 }
+
+TEST_F(IndexEstimatorTest, test_serialize_compression) {
+  std::vector<uint64_t> toInsert(10000);
+  constexpr uint64_t seq = 42;
+
+  auto buildEstimator = [&]() {
+    uint64_t i = 0;
+    auto est = std::make_unique<RocksDBCuckooIndexEstimatorType>(2048);
+    std::generate(toInsert.begin(), toInsert.end(), [&i] { return i++; });
+    for (auto it : toInsert) {
+      est->insert(it);
+    }
+
+    est->setAppliedSeq(seq);
+    
+    return est;
+  };
+  
+  auto validateSerializedValue = [&](auto& est, std::string const& serialization) {
+    arangodb::velocypack::StringRef ref(serialization);
+    RocksDBCuckooIndexEstimatorType copy(ref);
+
+    // After serialization => deserialization
+    // both estimates have to be identical
+    EXPECT_EQ(est.nrUsed(), copy.nrUsed());
+    EXPECT_EQ(est.nrCuckood(), copy.nrCuckood());
+    EXPECT_EQ(est.computeEstimate(), copy.computeEstimate());
+    EXPECT_EQ(seq, copy.appliedSeq());
+
+    // Now let us remove the same elements in both
+    bool coin = false;
+    for (auto it : toInsert) {
+      if (coin) {
+        est.remove(it);
+        copy.remove(it);
+      }
+      coin = !coin;
+    }
+
+    // We cannot relibly check inserts because the cuckoo has a random factor
+    // Still all values have to be identical
+
+    EXPECT_EQ(est.nrUsed(), copy.nrUsed());
+    EXPECT_EQ(est.nrCuckood(), copy.nrCuckood());
+    EXPECT_EQ(est.computeEstimate(), copy.computeEstimate());
+  };
+
+  { 
+    // uncompressed
+    auto est = buildEstimator();
+
+    std::string serialization;
+    auto format = RocksDBCuckooIndexEstimatorType::SerializeFormat::UNCOMPRESSED;
+    est->serialize(serialization, seq, format);
+    ASSERT_EQ(24641, serialization.size());
+    ASSERT_EQ(format, serialization[sizeof(uint64_t)]);
+
+    validateSerializedValue(*est, serialization);
+  }
+
+  { 
+    // compressed
+    auto est = buildEstimator();
+
+    std::string serialization;
+    auto format = RocksDBCuckooIndexEstimatorType::SerializeFormat::COMPRESSED;
+    est->serialize(serialization, seq, format);
+    ASSERT_EQ(10056, serialization.size());
```

Comment:
This sizing is actually more relevant.
How "Stable" is the Snappy compression here? Does it guarantee identical sizes even if the ordering of chars in the Compressed string may be modified?
Or is the pseudo-randomness under enough control that the non-compressed string between two runs is guaranteed to be identical?
This will be a randomly red test if the size differs just because the ordering of execution in unrelated tests has changed. (end thereby the random generator seed).

Reply:
Okay i have checked, the seed is constant and defined in the class alone.
=> This test will be deterministic this way, unless we change the seed, which is fine. :+1: